From 5fd2b90aaf7f425aafa5f3589d26178f85236ebb Mon Sep 17 00:00:00 2001
From: Jack Lin <jack.lin@dniserver.dnish.net>
Date: Tue, 7 Dec 2010 15:46:29 +0800
Subject: [PATCH 100/137] sysctl: Add a /proc/sys/dni_netfilter/conntrack_refresh_support interface

When /proc/sys/dni_netfilter/conntrack_refresh_support set to 1:
        Refresh the timer of the FTP control session when the FTP data session is keeping transmitting
	data (It is all TCP protocol).

When /proc/sys/dni_netfilter/conntrack_refresh_support set to 0(default):
        nf_conntrack_proto_tcp.c works as normal Linux code.

DNI Common Module Patch Review Checklist
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Here are some basic things that I (Jonathan Chang) have checked
in order to ensure the quality of common modules.

1: The syntax and comment are compliant.

2: No memory leak.
   * memory allocated by malloc() will be freed
   * memory allocated by calloc() will be freed

3: No deadlock such as
   * No function call deadlock.
   * No semaphore deadlock.
   * No tag or label deadlock.

4: No stack overflow caused by
   * infinite recursion
   * too large local array variables

5: coding completeness, such as
   * conditions completeness
   * functions completeness
   * update completeness
   * memory resources completeness
   * sessions resources completeness
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Signed-off-by: Jack Lin <jack.lin@dniserver.dnish.net>
Signed-off-by: Jonathan Chang <jonathan@dniserver.dnish.net>
---
 net/netfilter/dni_netfilter_sysctl.c |   11 +++++++++++
 1 files changed, 11 insertions(+), 0 deletions(-)

diff --git a/net/netfilter/dni_netfilter_sysctl.c b/net/netfilter/dni_netfilter_sysctl.c
index ef49da4..54f99bb 100644
--- a/net/netfilter/dni_netfilter_sysctl.c
+++ b/net/netfilter/dni_netfilter_sysctl.c
@@ -7,7 +7,18 @@
 #ifdef CONFIG_SYSCTL
 #include <linux/sysctl.h>
 
+extern int sysctl_conntrack_refresh_support;
+
 ctl_table dni_table[] = {
+	{
+		.ctl_name	= CTL_UNNUMBERED,
+		.procname	= "conntrack_refresh_support",
+		.data		= &sysctl_conntrack_refresh_support,
+		.maxlen		= sizeof(int),
+		.mode		= 0644,
+		.proc_handler	= proc_dointvec,
+		.strategy	= sysctl_intvec,
+	},
 	{0}
 };
 
-- 
1.7.4.1

